react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+export default class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      error: '',
+      iconClass: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleValidation = this.handleValidation.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ value: e.target.value });
+  }
+
+  handleValidation() {
+    if (this.state.value.length === 0) {
+      this.setState({
+        error: 'A password is required',
+        iconClass: 'fas fa-times icon-width'
+      });
+    } else if (this.state.value.length < 8) {
+      this.setState({
+        error: 'Password must be at least 8 digits',
+        iconClass: 'fas fa-times'
+      });
+    } else {
+      this.setState({
+        error: '',
+        iconClass: 'fas fa-check'
+      });
+    }
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const isValid = this.handleValidation();
+    if (isValid) {
```

Comment:
Because `handleValidation` doesn't return anything `isValid` will always be `undefined` so by making the above mentioned changes then this part will work properly.